### ISSUE TITLE
Wrong camera and lens set for lensfun when opening CR3 files

### ISSUE THREAD

**Issue by heckflosse** [Beep6581/RawTherapee#5950]
When opening a CR3 file, the wrong camera and lens are set for lensfun because lensfun search gets called with empty strings and returns the alphabetically first camera and lens.
![grafik](https://user-images.githubusercontent.com/13733487/95853997-93072680-0d56-11eb-891c-ef66ccb0676f.png)

This fixes it:
```diff
diff --git a/rtengine/rtlensfun.cc b/rtengine/rtlensfun.cc
index 363819cf7..58f3d12a3 100644
--- a/rtengine/rtlensfun.cc
+++ b/rtengine/rtlensfun.cc
@@ -438,7 +438,7 @@ std::vector<LFLens> LFDatabase::getLenses() const
 LFCamera LFDatabase::findCamera(const Glib::ustring &make, const Glib::ustring &model) const
 {
     LFCamera ret;
-    if (data_) {
+    if (data_ && !make.empty()) {
         MyMutex::MyLock lock(lfDBMutex);
         auto found = data_->FindCamerasExt(make.c_str(), model.c_str());
         if (found) {
@@ -453,7 +453,7 @@ LFCamera LFDatabase::findCamera(const Glib::ustring &make, const Glib::ustring &
 LFLens LFDatabase::findLens(const LFCamera &camera, const Glib::ustring &name) const
 {
     LFLens ret;
-    if (data_) {
+    if (data_ && !name.empty()) {
         MyMutex::MyLock lock(lfDBMutex);
         auto found = data_->FindLenses(camera.data_, nullptr, name.c_str());
         for (size_t pos = 0; !found && pos < name.size(); ) {

```

**Comment by Thanatomanic** [comment]
Looks good to push immediately 👍

**Comment by heckflosse** [comment]
Fixed with a30dca0b0e0e66cc9fa4363cdc22b07961340ee4
